cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java

```diff
@@ -164,7 +164,13 @@ private final void reconcileClusterCa(Reconciliation reconciliation, Handler<Asy
                     log.debug("{}: End generating cluster CA {}", reconciliation, clusterCaName);
                 } else {
                     log.debug("{}: The cluster CA {} already exists", reconciliation, clusterCaName);
-                    future.complete();
+
+                    SecretCertProvider secretCertProvider = new SecretCertProvider();
+                    Secret secret = secretCertProvider.createSecret(reconciliation.namespace(), clusterCaName,
+                            clusterCa.getData(), labels.toMap());
+
+                    secretOperations.reconcile(reconciliation.namespace(), clusterCaName, secret)
```

Comment:
In both path (if `clusterCa` is `null` or not), you need a `SecretCertProvider` instance and you are calling the `secretOperations.reconcile` with same parameters. We could move those lines after the `else` I guess and the `Secret` declaration (`null` initialized) before the `if`. Wdyt?